javascript/est-widget.js

```diff
@@ -28,24 +28,56 @@
 
     function main() {
         jQuery(document).ready(function ($) {
-            $('#est-widget-container').html(
-                '<input type="text" id="est-query">' +
-                '<input type="button" id="est-widget-search" value="search">' +
-                '<div id="est-widget-result"></div>'
-            );
-            $('#est-widget-search').on('click', function (e) {
-                estLoadData($('#est-query').val());
-            });
-
-            var estURL = "https://api.govwizely.com/environmental_solutions/search";
-
-            function estLoadData(search) {
-                var queryString = search ? '?q='+search : '';
-                $.getJSON(estURL + queryString, function (data) {
-                    $('#est-widget-result').html(JSON.stringify(data));
+            $.fn.est_widgetize = function (options) {
+                var apiKey = options['api_key'];
+                var widgetElementId = $(this).attr('id');
+
+                $('#' + widgetElementId).html(
+                    '<input type="text" id="est-query">' +
+                    '<input type="button" id="est-widget-search" value="search">' +
+                    '<pre id="est-widget-result"></pre>'
+                );
+
+                $('#est-widget-search').on('click', function (e) {
+                    estLoadData($('#est-query').val());
                 });
-            }
+
+                var estURL = "https://api.govwizely.com/v2/environmental_solutions/search";
+
+                function estLoadData(search) {
+                    $.getJSON(composeURL(search), function (data) {
+                        $('#est-widget-result').html(syntaxHighlight(data));
+                    });
+                }
+
+                function composeURL(search) {
+                    return estURL + '?api_key=' + apiKey + '&q=' + search
```

Comment:
If `search` is an empty string, you cannot add `&q=` to the URL. The API doesn't behave as intended if you do. See the difference:

https://api.govwizely.com/environmental_solutions/search
https://api.govwizely.com/environmental_solutions/search?q=

I have [opened a bug](https://www.pivotaltracker.com/story/show/97957276) reporting this, but I think in general it's good to not pass query params when we're not actually doing a filtered search.